spec/controllers/concerns/two_factor_authenticatable_methods_spec.rb

```diff
@@ -134,6 +134,28 @@
               )
             end
           end
+
+          context 'sign_in_notification_timeframe_expired but outside of max session timeframe' do
```

Comment:
I like knowing that this is handled gracefully, but my impression is that we're never expecting this to be possible, and that the user would either have `sign_in_new_device_at`, or they'd have a `sign_in_notification_timeframe_expired` event in the same active session.

Which is all to say that ideally we wouldn't need extra test coverage and the changes would fit as a refactor / optimization within existing test coverage.

But it'd be worth double-checking those assumptions.